x-pack/libbeat/persistentcache/persistentcache.go

```
@@ -97,7 +104,7 @@ func (c *PersistentCache) Get(k string, v interface{}) error {
 		return err
 	}
 	if c.refreshOnAccess && c.timeout > 0 {
-		c.store.Set([]byte(k), d, c.timeout)
+		_ = c.store.Set([]byte(k), d, c.timeout)
```

[Suggestion]
Shouldn't the error be logged? 
I see the original code also ignored the error, but I wonder if it's the best approach.
I know it's a cache and by definition it should not be a fatal error not being able to set a value in the cache, but if for some reason it keeps happening, it might impact the overall performance of whatever is using the cache. Besides if it fails often, there might be an issue that needs to be investigated.

---

i'm not sure, I wasn't intending to change the code (had to due to linter complains). Do you want me to add logging of the error?

---

> i'm not sure, I wasn't intending to change the code (had to due to linter complains).

This error breaks the contract, as it's clearly states the _last access time of the element is updated_. I only see one package using the cache. To avoid much changes, what do you think of the following. If you can confirm there is not "keyNotFoundError", you could log it as debug level or warning. regardless, would you mind updating the doc to say it might fail to update the _last access time_ and still return a nil error?

---

so currently (at least how I've implemented this) if key doesn't exist the store will return nil error and nil value (as is indicated by this test - https://github.com/stefans-elastic/beats/blob/drop-dbadger-io/x-pack/libbeat/kv/bbolt/bbolt_test.go#L100-L107). So i wonder - do I need to return an error in case of key not being find in store?

---

(also added a debug log)

---

(and updated godoc comment (i hope that's what you meant by "would you mind updating the doc to say it might fail to update the last access time and still return a nil error?"))